rohmu/object_storage/base.py

```diff
@@ -202,7 +205,30 @@ def copy_file(
         cannot be copied with this method. If no metadata is given copies the existing metadata."""
         raise NotImplementedError
 
-    def copy_files_from(self, *, source: BaseTransfer[SourceStorageModelT], keys: Collection[str]) -> None:
+    def copy_files_from(
+        self,
+        *,
+        source: BaseTransfer[SourceStorageModelT],
+        keys: Collection[str],
+        progress_fn: ObjectTransferProgressCallbackType = None,
+    ) -> None:
+        if isinstance(source, self.__class__):
+            for key in keys:
+                self._copy_file_from_bucket(source_bucket=source, source_key=key, destination_key=key, timeout=15)
+                if progress_fn is not None:
+                    progress_fn(key)
```

Comment:
I don't think the progress callback should take the key name as the parameter :

That prevents backward compatible implementation changes/optimization where the progress might jump suddenly by multiple files (if we do batches, etc.) and it would be silly to call the callback 100 times after a batch of 100 uploads, just to have the callback increment a counter by 1, 100 times.

I think a `progress_fn(completed_files: int, total_files: int)` is a better interface here, and happens to be more similar to what we have in other places for bytes, and more aligned with the needs of the library users.

Reply:
`progress_fn(completed_files: int, total_files: int)` is a better interface in terms of counting. The case that that we might skip or jump (bulk) a transfer is valid.

Initial idea
- provide more info about the actual transferred object
- but let the caller calculate the metrics